### ISSUE TITLE
recordPaparazzi on Java 17 - UninitializedPropertyAccessException lateinit property sessionParamsBuilder

### ISSUE THREAD

**Issue by mshearer123** [cashapp/paparazzi#1031]
**Description**
Error when running paparazzi record since updating to gradle 8 / Java 17
```
 kotlin.UninitializedPropertyAccessException: lateinit property sessionParamsBuilder has not been initialized
        at app.cash.paparazzi.Paparazzi$Companion.getSessionParamsBuilder$paparazzi(Paparazzi.kt:626)
        at app.cash.paparazzi.Paparazzi.prepare(Paparazzi.kt:162)
        at app.cash.paparazzi.Paparazzi$apply$statement$1.evaluate(Paparazzi.kt:124)
        at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
```

have added toolchains workaround
```
tasks.withType(Test).configureEach {
    javaLauncher = javaToolchains.launcherFor {
        languageVersion = JavaLanguageVersion.of(11)
    }
}
```

**Steps to Reproduce**
Unable to reproduce in sample project, but will keep trying over the next few nights.

**Expected behavior**
Expect to be able to run record tasks as before.

**Additional information:**
- Paparazzi Version: 1.3.1 / 1.3.2-snapshot
- OS: macOS 13.5 (aarch64)
- Compile SDK:
- Gradle Version: 33
- Android Gradle Plugin Version: 8.0.2

Note: Problem doesn't appear on our CI builds, on Linux

**Comment by oleksandrbalan-etn** [comment]
We have are integrated a paparazzi to our project and it works well for majority, but several of my colleagues have the same issue `kotlin.UninitializedPropertyAccessException: lateinit property sessionParamsBuilder has not been initialized`. 

@mshearer123 Did you manage to solve it? Maybe some workaround?

#### Additional information:

* Paparazzi Version: 1.3.1
* Compile SDK: 33
* Gradle Version: 8.1.1
* Android Gradle Plugin Version: 8.0.2

**Comment by mshearer123** [comment]
@oleksandrbalan-etn I'm afraid not, we're still stuck.  Haven't been able to reproduce in a sample project.

**Comment by rahnamaMo** [comment]
We have a similar situation. I was digging a little bit into it the other day and I found something pretty weird; we all have apple silicon macs, and while checking all the logs I realised the library is trying to load layoutlib in a path that is for x86 intel macs. I debugged, and found out that it indeed considers my system as an intel macbook. I have no idea what's happening to be honest since the logic there seems to be pretty straightforward, but may be worth looking into.
```
private fun getNativeLibDir(): String {
    val osName = System.getProperty("os.name").toLowerCase(Locale.US)
    val osLabel = when {
      osName.startsWith("windows") -> "win"
      osName.startsWith("mac") -> {
        val osArch = System.getProperty("os.arch").lowercase(Locale.US)
        if (osArch.startsWith("x86")) "mac" else "mac-arm"
      }
      else -> "linux"
    }
    return "$osLabel/lib64"
  }
  ```

**Comment by rahnamaMo** [comment]
Okay, so I fixed this issue for our case, here what's I found and how I fixed it at last:

**TLDR;** Your Android Studio is probably running the tests on JRE 11. Make sure your JDK 17 is aarch64 native and reinstall Android Studio completely.

**TL;** As you probably know `System.getProperty("os.arch")` returns the configuration of the system that JRE is being run on, not your actual system. I added `System.getProperty("os.arch")` into one of my simple non-paparazzi unit tests and the returning value was indeed `x86_64` instead of `aarch64`. So at this point I was sure there's nothing wrong with paparazzi specifically.

Next, I made sure I have the right version of java with the right architecture, to become 100% visually sure, I downloaded a Java 17 JDK from [here](https://adoptium.net/temurin/releases/?variant=openjdk17&jvmVariant=hotspot&os=mac&package=jdk&arch=aarch64) for aarch64 and install it. updated my JAVA_HOME and everything, and made sure my project in Android Studio is also using the correct JDK. Re-ran the test and still I was getting `x86_64` instead of `aarch64`. It's worth mentioning that there was no difference between running the tests via terminal or via AS GUI. I even went one step further and deleted all other installed JDKs I had, still no luck.

But this was when I realised Android Studio is running my tests [miraculously] on JRE 11. Since all previous "invalidate cache and restart" attempts did not have positive result, this time I just uninstalled and reinstalled Android Studio and finally after that it was working as it should have.

**Comment by rahnamaMo** [comment]
@mshearer123 sure thing:

- Compile SDK Version: 33
- Build Tools Version: 31.0.0
- Gradle Version: 8.1.1
- Android Gradle Plugin Version: 8.0.1
- Paparazzi Version: checked on both 1.2.0 and 1.3.1
- Kotlin: 1.8.10
- Groovy: 3.0.15
- Ant: Apache Ant(TM) version 1.10.11 compiled on July 10 2021
- JVM: 17.0.8 (Eclipse Adoptium 17.0.8+7)
- OS: Mac OS X 13.5 aarch64
- Android Studio Giraffe | 2022.3.1 Patch 1

**Comment by rahnamaMo** [comment]
@oleksandrbalan-etn that's amazing news 🥳 So glad I could help.

**Comment by mshearer123** [comment]
Same, once all other JDKS were removed.

thanks @rahnamaMo @oleksandrbalan-etn

**Comment by iamutkarshtiwari** [comment]
I am facing the same issue. 
I am running paparazzi in a Github action with the following config:
```yaml
      - name: Set up JDK 17
        uses: actions/setup-java@v3
        with:
          distribution: 'temurin'
          java-version: '17'
          architecture: ''
          overwrite-settings: false
      - name: Check out code
        uses: actions/checkout@v3
        with:
          fetch-depth: 0
      - name: Run screenshot tests
        id: run-screenshot-tests
        run: ./gradlew verifyPaparazziDebug
```

But it fails with the same error

**Comment by shoaibmushtaq25** [comment]
Since, I am using MAC with Apple chip.
I had the same issue. Uninstalling all JDKs with x86_64 and keeping arm64 JDKs works for me.
Thanks for the insights @rahnamaMo 👍

**Comment by jrodbx** [comment]
Closing as wontfix, since this is a user environment issue, and there's a suitable workaround (thanks @rahnamaMo!)
